File: frontend/src/types/order.ts
// 订单响应类型
export interface OrderResponse {
  id: number
  platformOrderNo: string
  customerId: number
  customerName?: string
  salesId: number
  salesUserName?: string
  supplierCompanyId?: number
  supplierCompanyName?: string
  orderType: string
  orderStatus: string
  orderSource: string
  expectedDeliveryDate: string
  approvalStatus: string
  approvalComment?: string
  approvalAt?: string
  approvalBy?: number
  totalAmount: string
  discountAmount: string
  taxAmount: string
  shippingAmount: string
  finalAmount: string
  currency: string
  paymentStatus: string
  paymentMethod: string
  paymentAt?: string
  deliveryAddress: string
  deliveryContact: string
  deliveryPhone: string
  specialRequirements?: string
  remarks?: string
  sourceOrderId?: string
  aiConfidence?: string
  aiProcessed: boolean
  extendedFields?: Record<string, any>
  createdAt: string
  updatedAt: string
  deleted: boolean
  orderItems: OrderItemResponse[]
}

// 订单项响应类型
export interface OrderItemResponse {
  id: number
  orderId: number
  productId: number
  productName?: string
  productSku: string
  productSpecifications?: string
  quantity: number
  unitPrice: string
  unit: string
  discountRate?: string
  discountAmount?: string
  taxRate?: string
  taxAmount?: string
  subtotal: string
  aiMappedProductId?: number
  aiConfidence?: number
  aiProcessingStatus?: string
  aiProcessingResult?: string
  createdAt: string
  updatedAt: string
}

// 创建订单请求类型
export interface OrderCreateRequest {
  // 后端必需字段
  salesOrderId: string
  salesUserId: number
  salesCompanyId: number
  customerId: number
  source: string
  templateVersion?: string
  remarks?: string
  extendedFields?: Record<string, any>
  orderItems: OrderItemCreateRequest[]
  
  // 前端显示字段（不发送到后端）
  orderType?: string
  expectedDeliveryDate?: string
  currency?: string
  specialRequirements?: string
  deliveryAddress?: string
  deliveryContact?: string
  deliveryPhone?: string
  discountAmount?: string
  shippingAmount?: string
  taxAmount?: string
  finalAmount?: string
  paymentMethod?: string
  platformOrderNo: string
  sourceOrderId?: string
}

// 更新订单请求类型
export interface OrderUpdateRequest {
  customerId?: number
  remarks?: string
  extendedFields?: Record<string, any>
  orderItems?: OrderItemUpdateRequest[]
}

// 更新订单项请求类型
export interface OrderItemUpdateRequest {
  id?: number
  productId: number
  quantity: number
  unitPrice: string
  remarks?: string
  extendedFields?: Record<string, any>
}

// 创建订单项请求类型
export interface OrderItemCreateRequest {
  // 后端必需字段
  productId: number
  quantity: number
  unitPrice: string
  remarks?: string
  extendedFields?: Record<string, any>
  
  // 前端显示字段（不发送到后端）
  productName?: string
  productSku?: string
  productSpecifications?: string
  unit?: string
  subtotal?: string
}

// 订单列表请求类型
export interface OrderListRequest {
  current: number
  size: number
  platformOrderNo?: string
  customerName?: string
  sourceOrderNo?: string
  orderStatus?: string
  source?: string
  dateRange?: string[]
  // 后端期望的字段
  status?: string
  createdAtStart?: string
  createdAtEnd?: string
  salesUserId?: number
  supplierCompanyId?: number
  customerId?: number
}

// 订单状态更新请求类型
export interface OrderStatusUpdateRequest {
  targetStatus: string      // 目标状态
  reason?: string          // 状态变更原因
  remarks?: string         // 状态变更备注
  operatorId: number       // 操作人ID (必填)
  operatorName?: string    // 操作人姓名
  operatorRole?: string    // 操作人角色
  forceUpdate?: boolean    // 是否强制更新
}

// 订单冲突解决请求类型
export interface OrderConflictResolutionRequest {
  resolutionType: 'REJECT' | 'MANUAL_CONFIRM' | 'AUTO_MERGE'
  selectedOrderId?: number
  comment?: string
}

// 订单批量处理请求类型
export interface OrderBatchProcessRequest {
  orderIds: number[]
  action: 'UPDATE_STATUS' | 'DELETE' | 'ASSIGN_SUPPLIER'
  data?: any
}

// 订单状态日志类型
export interface OrderStatusLog {
  id: number
  orderId: number
  fromStatus?: string        // 变更前状态
  toStatus: string          // 变更后状态（对应前端的orderStatus）
  changeReason?: string
  operatorType: string
  operatorId?: number
  operatorName?: string
  operatorRole?: string
  operatedAt?: string
  isSystemOperation: boolean
  operationSource: string
  remarks?: string
  createdAt: string
  updatedAt?: string
}

// 订单审批日志类型
export interface OrderApprovalLog {
  id: number
  orderId: number
  action: string
  status: string
  comment?: string
  approverId: number
  approvalAt: string
  createdAt: string
  approverName?: string
  approverRole?: string
}

// 物流信息类型
export interface LogisticsInfo {
  id: number
  orderId: number
  trackingNumber: string
  carrier: string
  carrierCode?: string
  shippingMethod: string
  shippingDate?: string
  estimatedDeliveryDate?: string
  actualDeliveryDate?: string
  shippingAddress: string
  deliveryAddress: string
  shippingContact: string
  deliveryContact: string
  shippingPhone: string
  deliveryPhone: string
  shippingNotes?: string
  deliveryNotes?: string
  status: string
  warehouseId?: number
  warehouseName?: string
  createdAt: string
  updatedAt: string
  platformOrderId?: string
}

// Excel导入日志类型
export interface ExcelImportLog {
  id: number
  fileName: string
  fileSize: number
  fileHash: string
  importStatus: string
  totalRows: number
  processedRows: number
  successRows: number
  failedRows: number
  errorMessage?: string
  aiProcessingStatus?: string
  aiProcessingResult?: string
  importUserId: number
  startedAt: string
  completedAt?: string
  createdAt: string
  updatedAt: string
  extendedFields?: string
  importUserName?: string
}

// 订单状态枚举
export enum OrderStatus {
  DRAFT = 'DRAFT',
  SUBMITTED = 'SUBMITTED',
  SUPPLIER_CONFIRMED = 'SUPPLIER_CONFIRMED',
  SHIPPED = 'SHIPPED',
  IN_TRANSIT = 'IN_TRANSIT',
  DELIVERED = 'DELIVERED',
  COMPLETED = 'COMPLETED',
  CANCELLED = 'CANCELLED',
  REJECTED = 'REJECTED'
}

// 订单来源枚举
export enum OrderSource {
  MANUAL = 'MANUAL',
  EXCEL_IMPORT = 'EXCEL_IMPORT',
  API = 'API',
  WEBSITE = 'WEBSITE',
  IMAGE_RECOGNITION = 'IMAGE_RECOGNITION',
  PDF_EXTRACTION = 'PDF_EXTRACTION',
  WORD_EXTRACTION = 'WORD_EXTRACTION',
  EMAIL_EXTRACTION = 'EMAIL_EXTRACTION',
  VOICE_RECOGNITION = 'VOICE_RECOGNITION'
}

// 订单类型枚举
export enum OrderType {
  NORMAL = 'NORMAL',
  URGENT = 'URGENT',
  SAMPLE = 'SAMPLE',
  REPAIR = 'REPAIR'
}

// 支付状态枚举
export enum PaymentStatus {
  PENDING = 'PENDING',
  PARTIAL = 'PARTIAL',
  COMPLETED = 'COMPLETED',
  FAILED = 'FAILED',
  REFUNDED = 'REFUNDED'
}

// 支付方式枚举
export enum PaymentMethod {
  BANK_TRANSFER = 'BANK_TRANSFER',
  CASH = 'CASH',
  CHECK = 'CHECK',
  CREDIT_CARD = 'CREDIT_CARD',
  OTHER = 'OTHER'
}

// 审批状态枚举
export enum ApprovalStatus {
  PENDING = 'PENDING',
  APPROVED = 'APPROVED',
  REJECTED = 'REJECTED',
  NOT_REQUIRED = 'NOT_REQUIRED'
}

File: backend/src/main/java/com/yibang/erp/domain/entity/Order.java
package com.yibang.erp.domain.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.experimental.Accessors;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.Map;

/**
 * 订单实体 - 基于现有数据库表结构
 */
@Data
@EqualsAndHashCode(callSuper = false)
@Accessors(chain = true)
@TableName("orders")
public class Order {

    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    /**
     * 平台订单号（系统内部唯一标识）
     */
    @TableField("platform_order_id")
    private String platformOrderId;

    /**
     * 客户ID
     */
    @TableField("customer_id")
    private Long customerId;

    /**
     * 销售ID
     */
    @TableField("sales_id")
    private Long salesId;

    /**
     * 供应链公司ID
     */
    @TableField("supplier_company_id")
    private Long supplierCompanyId;

    /**
     * 订单类型
     */
    @TableField("order_type")
    private String orderType; // NORMAL, WHOLESALE, RUSH, SAMPLE

    /**
     * 订单来源
     */
    @TableField("order_source")
    private String orderSource; // MANUAL, EXCEL_IMPORT, API, WEBSITE

    /**
     * 订单状态
     */
    @TableField("order_status")
    private String orderStatus;

    /**
     * 审核状态
     */
    @TableField("approval_status")
    private String approvalStatus; // PENDING, APPROVED, REJECTED

    /**
     * 审核意见
     */
    @TableField("approval_comment")
    private String approvalComment;

    /**
     * 审核时间
     */
    @TableField("approval_at")
    private LocalDateTime approvalAt;

    /**
     * 审核人ID
     */
    @TableField("approval_by")
    private Long approvalBy;

    /**
     * 订单总金额
     */
    @TableField("total_amount")
    private BigDecimal totalAmount;

    /**
     * 折扣金额
     */
    @TableField("discount_amount")
    private BigDecimal discountAmount;

    /**
     * 税额
     */
    @TableField("tax_amount")
    private BigDecimal taxAmount;

    @TableField("logistics_order_number")
    private String logisticsOrderNumber;

    /**
     * 物流公司
     */
    @TableField("logistics_company")
    private String logisticsCompany;

    /**
     * 运费
     */
    @TableField("shipping_amount")
    private BigDecimal shippingAmount;

    /**
     * 最终金额
     */
    @TableField("final_amount")
    private BigDecimal finalAmount;

    /**
     * 货币代码
     */
    @TableField("currency")
    private String currency;

    /**
     * 支付状态
     */
    @TableField("payment_status")
    private String paymentStatus; // UNPAID, PARTIALLY_PAID, PAID, REFUNDED

    /**
     * 支付方式
     */
    @TableField("payment_method")
    private String paymentMethod;

    /**
     * 支付时间
     */
    @TableField("payment_at")
    private LocalDateTime paymentAt;

    /**
     * 收货地址
     */
    @TableField("delivery_address")
    private String deliveryAddress;

    /**
     * 收货联系人
     */
    @TableField("delivery_contact")
    private String deliveryContact;

    /**
     * 收货联系电话
     */
    @TableField("delivery_phone")
    private String deliveryPhone;

    /**
     * 预计交货日期
     */
    @TableField("expected_delivery_date")
    private LocalDate expectedDeliveryDate;

    /**
     * 特殊要求
     */
    @TableField("special_requirements")
    private String specialRequirements;

    /**
     * AI处理置信度
     */
    @TableField("ai_confidence")
    private BigDecimal aiConfidence;

    /**
     * 是否经过AI处理
     */
    @TableField("ai_processed")
    private Boolean aiProcessed;



    /**
     * 创建时间
     */
    @TableField("created_at")
    private LocalDateTime createdAt;

    /**
     * 更新时间
     */
    @TableField("updated_at")
    private LocalDateTime updatedAt;

    /**
     * 创建人ID
     */
    @TableField("created_by")
    private Long createdBy;

    /**
     * 更新人ID
     */
    @TableField("updated_by")
    private Long updatedBy;

    @TableField("province_name")
    private String provinceName;


    @TableField("city_name")
    private String cityName;

    @TableField("district_name")
    private String  districtName;

    @TableField("source_order_id")
    private String sourceOrderId;





    /**
     * 是否删除
     */
    @TableField("deleted")
    @TableLogic
    private Boolean deleted;

    // 扩展字段（用于存储动态字段，不映射到数据库）
    @TableField(exist = false)
    private Map<String, Object> extendedFields;

    // 扩展字段（用于存储动态字段，不映射到数据库）
    @TableField(exist = false)
    private String salesOrderId; // 销售订单ID，用于前端显示

    // 扩展字段（用于存储动态字段，不映射到数据库）
    @TableField(exist = false)
    private String templateVersion; // 模板版本，用于前端显示

    // 扩展字段（用于存储动态字段，不映射到数据库）
    @TableField(exist = false)
    private String remarks; // 订单备注，用于前端显示

    /**
     * 获取扩展字段Map
     */
    public Map<String, Object> getExtendedFieldsMap() {
        if (extendedFields == null) {
            return Map.of();
        }
        return extendedFields;
    }

    /**
     * 设置扩展字段Map
     */
    public void setExtendedFieldsMap(Map<String, Object> fields) {
        this.extendedFields = fields;
    }
}
